File: spring-email-java/src/main/java/com/github/migangqui/spring/email/service/EmailSender.java
package com.github.migangqui.spring.email.service;

import com.github.migangqui.spring.email.model.SendEmailResult;
import com.github.migangqui.spring.email.model.Email;

import java.util.concurrent.Future;

public interface EmailSender {
    SendEmailResult send(Email email);
    Future<SendEmailResult> sendAsync(Email email);
}

File: spring-email-kotlin/src/main/kotlin/com/github/migangqui/spring/email/service/EmailSender.kt
package com.github.migangqui.spring.email.service

import com.github.migangqui.spring.email.model.Email
import com.github.migangqui.spring.email.model.SendEmailResult
import java.util.concurrent.Future

interface EmailSender {
    fun send(email: Email): SendEmailResult
    fun sendAsync(email: Email): Future<SendEmailResult>
}